weaviate/collections/batch/batch_wrapper.py

```diff
@@ -26,26 +27,19 @@ def __init__(
         self._concurrent_requests: int = 2
 
         self._batch_data = _BatchDataWrapper()
-        self.__shut_background_thread_down: Optional[threading.Event] = None
 
     def __start_event_loop_thread(self, loop: asyncio.AbstractEventLoop) -> None:
-        while (
-            self.__shut_background_thread_down is not None
-            and not self.__shut_background_thread_down.is_set()
-        ):
-            if loop.is_running():
-                continue
-            else:
-                loop.run_forever()
-
-    def _open_async_connection(self) -> asyncio.AbstractEventLoop:
+        loop.set_debug(True)  # in case of errors, shows async errors in the thread to users
+        loop.run_forever()
+
+    def _start_event_loop(self) -> asyncio.AbstractEventLoop:
         try:
             self._current_loop = asyncio.get_running_loop()
+            nest_asyncio.apply(self._current_loop)
```

Comment:
why only apply in this branch?

Reply:
In environments where there is a running event loop in the main thread, e.g. an async API or Py notebook, we need to use this monkey patching library to allow the loop to be re-ran again within the event loop thread

Reply:
I did some reading on `nest_asyncio` to be sure our usage is sound: https://github.com/erdewit/nest_asyncio/issues/36, and also tested running batching concurrently with an `async` GET in a FastAPI scratch app and it works well